routing/router_test.go

```diff
@@ -4260,6 +4270,9 @@ func TestSendMPPaymentFailedWithShardsInFlight(t *testing.T) {
 	// methods are called as expected.
 	require.Error(t, err, "expected send payment error")
 	require.EqualValues(t, [32]byte{}, p, "preimage not match")
+	require.GreaterOrEqual(t, getPaymentResultCnt, 1)
```

Comment:
I think we need to check `getPaymentResultCnt` is equal to 4 to make sure we know all the attempts are failed.

Reply:
Yeah right, just reminded me that it's difficult to be deterministic here. I guess as long as `getPaymentResultCnt == cntFail` we are good here.